torch/_dynamo/variables/torch.py

```diff
@@ -782,26 +782,32 @@ def speculate_subgraph(
     tx, f, sub_args, graph_checkpoint, checkpoint, *, always_restore=False
 ):
     from . import AutogradFunctionContextVariable, ConstantVariable, TensorVariable
+    from .builder import wrap_fx_proxy
 
     try:
         with tx.output.new_subtracer() as tracer:
             args = []
             # One argument to graph per sub_args
             for a in sub_args:
                 if a is None:
-                    a = ConstantVariable(None)
+                    new_arg = ConstantVariable(None)
                 assert not isinstance(
                     a, torch.Tensor
                 ), "Tensors should already be tracked?"
                 if isinstance(a, ConstantVariable):
                     proxy = tracer.create_graph_input("const")
+                    new_arg = a
```

Comment:
What is going on here? if `a` is None, then we do not create an input for it in the subgraph?

Also, is it possible to add a test?